plugins/modules/helm_cli_info.py

```diff
@@ -0,0 +1,289 @@
+                    return release
+        else:
+            for release in state:
+                if release['name'] == release_name and release['namespace'] == release_namespace:
+                    return release
+    return None
+
+
+# Get Release state from deployed release
+def get_release_status(command, release_name, release_namespace):
+    list_command = command + " list --output=yaml "
+
+    if not is_helm_2:
+        list_command += " --namespace=" + release_namespace
+        list_command += " --filter "
+
+    list_command += release_name
+
+    rc, out, err = module.run_command(list_command)
+
+    if rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=list_command
+        )
+
+    release = get_release(yaml.safe_load(out), release_name, release_namespace)
+
+    if release is None:  # not install
+        return None
+
+    release['values'] = get_values(command, release_name, release_namespace)
+
+    return release
+
+
+def main():
+    global module, is_helm_2
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            binary_path=dict(type='path'),
+            release_name=dict(type='str', required=True, aliases=['name']),
+            release_namespace=dict(type='str', required=True, aliases=['namespace']),
+            tiller_host=dict(type='str'),
+            tiller_namespace=dict(type='str', default='kube-system'),
+
+            # Helm options
+            kube_context=dict(type='str'),
+            kubeconfig_path=dict(type='path', aliases=['kubeconfig']),
+        ),
+        supports_check_mode=True,
+    )
+    if not HAS_YAML:
+        module.fail_json(msg="Could not import the yaml python module. Please install `yaml` package.")
+
+    bin_path = module.params.get('binary_path')
+    release_name = module.params.get('release_name')
+    release_namespace = module.params.get('release_namespace')
+    tiller_host = module.params.get('tiller_host')
+    tiller_namespace = module.params.get('tiller_namespace')
+
+    # Helm options
+    kube_context = module.params.get('kube_context')
+    kubeconfig_path = module.params.get('kubeconfig_path')
+
+    if bin_path is not None:
+        helm_cmd_common = bin_path
+    else:
+        helm_cmd_common = module.get_bin_path('helm', required=True)
+
+    is_helm_2 = get_helm_client_version(helm_cmd_common)
+
+    # Helm 2 need tiller, Helm 3 and higher doesn't
+    if is_helm_2:
+        if tiller_host is not None:
+            helm_cmd_common += " --host=" + tiller_namespace
+        else:
+            helm_cmd_common += " --tiller-namespace=" + tiller_namespace
+
+    if kube_context is not None:
+        helm_cmd_common += " --kube-context " + kube_context
+
+    if kubeconfig_path is not None:
+        helm_cmd_common += " --kubeconfig " + kubeconfig_path
+
```

Comment:
Can we deprecate https://github.com/ansible-collections/community.general/blob/master/plugins/modules/cloud/misc/helm.py at the same time when we merge these modules?

Reply:
PR created: https://github.com/ansible-collections/community.general/pull/115